src/modules/common/selectors/availableLanguagesSelector.js

```diff
@@ -9,11 +9,15 @@ import type {
 const cityLanguagesSelector = (state: StateType): ?Array<string> => state.cityContent.languages &&
   state.cityContent.languages.map(languageModel => languageModel.code)
 
-const categoryRouteSelector = (state: StateType, ownProps): ?CategoryRouteStateType =>
-  state.cityContent.categoriesRouteMapping[ownProps.navigation.getParam('key')]
+const categoryRouteSelector = (state: StateType, ownProps): ?CategoryRouteStateType => {
+  const route = state.cityContent.categoriesRouteMapping[ownProps.navigation.getParam('key')]
+  return route.success && route
```

Comment:
I could be wrong, but there is no gurantee that sucess does exist. So this should throw a flow error.

Reply:
Why should this throw a flow error? If `route.success` does not exist, `route.success` is `undefined` and so it returns `undefined`, which is okay since the return type is `?CategoryRouteState?

Reply:
I thought that something like that does not work but it does:
```
type AaType = {|
  a: boolean
|}

type BbType = {|
  b: boolean
|}

type CcType = AaType | BbType

const c: CcType = {b: true}

if (c.a) {
  console.log()
}
```

So this is like a type check.